tests/integration_tests/reports/alert_tests.py

```diff
@@ -98,6 +99,86 @@ def test_execute_query_as_report_executor(
     app.config["ALERT_REPORTS_EXECUTE_AS"] = original_config
 
 
+def test_execute_query_mutate_query_enabled(
+    mocker: MockerFixture,
+    app_context: AppContext,
+    get_user,
+) -> None:
+    from superset.commands.report.alert import AlertCommand
+    from superset.reports.models import ReportSchedule
+
+    def mock_mutate_query(sql: str, **kwargs: Any) -> str:
+        return "before " + sql + " after"
+
+    app.config["SQL_QUERY_MUTATOR"] = mock_mutate_query
+    app.config["MUTATE_ALERT_QUERY"] = True
+    app.config["ALERT_REPORTS_EXECUTE_AS"] = [ExecutorType.OWNER]
+
+    mocker.patch("superset.commands.report.alert.override_user")
+    mock_df = mocker.MagicMock(spec=pd.DataFrame)
+    mock_df.empty = True
+    mock_database = get_example_database()
+    mock_get_df = mocker.patch.object(mock_database, "get_df", return_value=mock_df)
+    mock_limited_sql = mocker.patch.object(
+        mock_database, "apply_limit_to_sql", return_value="SELECT 1\nLIMIT 2"
+    )
+
+    report_schedule = ReportSchedule(
+        created_by=get_user("admin"),
+        owners=[get_user("admin")],
+        type=ReportScheduleType.ALERT,
+        description="description",
+        crontab="0 9 * * *",
+        creation_method=ReportCreationMethod.ALERTS_REPORTS,
+        sql="SELECT 1",
+        grace_period=14400,
+        working_timeout=3600,
+        database=mock_database,
+        validator_config_json='{"op": "==", "threshold": 1}',
+    )
+
+    AlertCommand(report_schedule=report_schedule, execution_id=uuid.uuid4()).run()
+    mock_get_df.assert_called_once_with(
+        sql=f"before {mock_limited_sql.return_value} after"
+    )
+
+
+def test_execute_query_mutate_query_disabled(
+    mocker: MockerFixture,
+    app_context: AppContext,
+    get_user,
+) -> None:
+    from superset.commands.report.alert import AlertCommand
+    from superset.reports.models import ReportSchedule
+
+    def mock_mutate_query(sql: str, **kwargs: Any) -> str:
+        return "before " + sql + " after"
+
+    app.config["SQL_QUERY_MUTATOR"] = mock_mutate_query
+    app.config["MUTATE_ALERT_QUERY"] = False
+    app.config["ALERT_REPORTS_EXECUTE_AS"] = [ExecutorType.OWNER]
```

Comment:
@Vitor-Avila I think you're going to want to reset these at the end of the test for cleanup, too.

Reply:
good catch! Changed the test logic a bit and re-applied the initial app configs at the end of them 🙌